### PR TITLE
Support Laravel 11.x

### PR DESCRIPTION
**Pull request by SupianIDz**
- Add PHP 8.2
- Support Laravel 11

### CODE REVIEW

**Comment by lloricode** [comment]
can you update GitHub action as well?

**Comment by SupianIDz** [comment]
> can you update GitHub action as well?

There are some dev dependencies that still haven't updated to PHP 8.2, I think it's better to wait a little longer for the merge.

At least, this solves the problem temporarily.

**Comment by lloricode** [comment]
hmm, also can you add php 8.3 on test? then exclude php8.1 on laravel 11

**Comment by SupianIDz** [comment]
Sure, PHP 8.1 is already excluded from Laravel 11.

**Review by lloricode — CHANGES_REQUESTED** [review]
please add php8.3 on test, thank you

**Review by lloricode — CHANGES_REQUESTED** [review]
please add php8.3, thank you

**Review by lloricode — APPROVED** [review]

**Comment by ariaieboy** [comment]
cc: @denisdulici

**Comment by andreeastrattner** [comment]
Would appreciate laravel 11 support. Currently, this package is a blocker.

**Comment by denisdulici** [comment]
@SupianIDz Please revert the PHP version change so I can merge the PR.

**Comment by SupianIDz** [comment]
@denisdulici done

**Comment by SupianIDz** [comment]
Do you mean stay on PHP 8.0? I thought only PHP 8.3 was reverted.

**Comment by SupianIDz** [comment]
Do you mean stay on PHP 8.0? I thought only PHP 8.3 was reverted.

**Comment by ariaieboy** [comment]
> Do you mean stay on PHP 8.0? I thought only PHP 8.3 was reverted.

You don't need to specify each PHP version on composer.json 
`^8.0|^8.1|^8.2` == `^8.0`

change composer.json PHP version to `^8.0` and add PHP 8.3 to test workflow file.

**Comment by SupianIDz** [comment]
> > Do you mean stay on PHP 8.0? I thought only PHP 8.3 was reverted.
> 
> You don't need to specify each PHP version on composer.json `^8.0|^8.1|^8.2` == `^8.0`
> 
> change composer.json PHP version to `^8.0` and add PHP 8.3 to test workflow file.

Ahh, I just found out after looking deeper, thanks for the insight.

**Comment by SupianIDz** [comment]
It has been updated, please check.

**Comment by ariaieboy** [comment]
@denisdulici

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,15 +10,19 @@ jobs:
 
     strategy:
       matrix:
-        php: ['8.0', '8.1', '8.2']
+        php: ['8.0', '8.1', '8.2', '8.3']
         laravel: [9.*, 10.*]
```
Comment (denisdulici) [inline .github/workflows/tests.yml:14]: Please add 11 here.

**Review by lloricode — CHANGES_REQUESTED** [review]
please squash all commits

**Comment by SupianIDz** [comment]
> please squash all commits

done.

----

@denisdulici PHP 8.3 has been added to Matrix, and excluding for PHP 8.0 & PHP 8.1 in Laravel 11.

But I saw that there was a test error related to "mock", I'm not very good at mocking.

Thank you for the assistance and new insights about versioning.

**Review by lloricode — APPROVED** [review]
thank you :beers:

**Comment by ariaieboy** [comment]
the `Pslam` version 4 is not compatible you should add `Pslam` version 5.

**Comment by patriktoth67** [comment]
When can we expect the laravel 11 integration? At the moment this package is the last one preventing me from upgrading.

**Comment by SupianIDz** [comment]
For those who want to need a quick solution while waiting.

```json
"repositories": [
    {
        "url": "https://github.com/SIOPENHSS/laravel-money.git",
        "type": "git"
    }
],
"require": {
     "akaunting/laravel-money": "master"
}
```

cc : @andreeastrattner @patriktoth67

**Comment by denisdulici** [comment]
Tests failed, so the current PR is not stable. I'll merge this https://github.com/akaunting/laravel-money/pull/98 PR and release a new version today.

**Pull request closed without merging**